src/NHibernate/IFutureValueAsync.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading.Tasks;
+
+namespace NHibernate
+{
+	public interface IFutureValueAsync<T>
+	{
+		Task<T> GetValue();
```

Comment:
True, since there is an async `ToList` callable without a `cancellationToken`... A bit unfortunate. We may instead have a `IFutureEnumerable<T> : IEnumerable<T>` exposing a additional `IAsyncEnumerable<T> AsyncEnumerable` property. Not that nice, but still avoiding to have two distinct synchronous calls to do depending on whether we want later asynchronously or synchronously resolve the future.